File: backend/app/schemas/freelancer_profile.py
from pydantic import BaseModel


class FreelancerLanguage(BaseModel):
    name: str
    level: str


class FreelancerBadges(BaseModel):
    successRate: float
    topRated: bool
    avgResponseTime: str


class FreelancerStats(BaseModel):
    completedProjects: int
    hoursWorked: int
    repeatClients: int
    memberSince: str
    lastDelivery: str


class ExperienceEntry(BaseModel):
    period: str
    role: str
    company: str
    description: str


class EducationEntry(BaseModel):
    period: str
    degree: str
    institution: str


class WorkMethodCategory(BaseModel):
    label: str
    items: list[str]


class FreelancerProfile(BaseModel):
    id: str
    name: str
    email: str
    initials: str
    rating: float
    reviewCount: int
    specialty: str
    category: str
    country: str
    countryCode: str
    timezone: str
    online: bool
    languages: list[FreelancerLanguage]
    skills: list[str]
    badges: FreelancerBadges
    stats: FreelancerStats
    bio: str
    differentiators: list[str]
    experience: list[ExperienceEntry]
    education: list[EducationEntry]
    tools: list[str]
    availabilityStatus: str
    workMethods: list[WorkMethodCategory]

File: frontend/src/types/freelancerProfile.ts
export interface FreelancerLanguage {
  name: string;
  level: string;
}

export interface FreelancerBadges {
  successRate: number;
  topRated: boolean;
  avgResponseTime: string;
}

export interface FreelancerStats {
  completedProjects: number;
  hoursWorked: number;
  repeatClients: number;
  memberSince: string;
  lastDelivery: string;
}

export interface ExperienceEntry {
  period: string;
  role: string;
  company: string;
  description: string;
}

export interface EducationEntry {
  period: string;
  degree: string;
  institution: string;
}

export interface WorkMethodCategory {
  label: string;
  items: string[];
}

export type WorkMethods = WorkMethodCategory[];

export interface FreelancerProfile {
  id: string;
  name: string;
  email: string;
  initials: string;
  rating: number;
  reviewCount: number;
  specialty: string;
  category: string;
  country: string;
  countryCode: string;
  timezone: string;
  online: boolean;
  languages: FreelancerLanguage[];
  skills: string[];
  badges: FreelancerBadges;
  stats: FreelancerStats;
  bio: string;
  differentiators: string[];
  experience: ExperienceEntry[];
  education: EducationEntry[];
  tools: string[];
  availabilityStatus: string;
  workMethods: WorkMethods;
}
